src/ext/browserUtils.js

```diff
@@ -64,18 +64,21 @@ function attemptRedeem(url, respTabId, target) {
 
 // Actually activate the redemption request
 function fireRedeem(url, respTabId, target) {
-    if (REDEEM_METHOD == "reload") {
+    if (REDEEM_METHOD === "reload") {
         setSpendFlag(url.host, true);
         let targetUrl = target[respTabId];
-        if (url.href == targetUrl) {
+        if (url.href === targetUrl) {
             chrome.tabs.update(respTabId, { url: targetUrl });
         } else {
             // set a reload in the future when the target has been inited, also
             // reset timer for resetting vars
             futureReload[respTabId] = url.href;
             timeSinceLastResp = Date.now();
         }
-    } else {
+    } else if (REDEEM_METHOD === "no-reload") {
+    //    pass
+    }
+    else {
         throw new Error("[privacy-pass]: Incompatible redeem method selected.");
```

Comment:
This "else if() { // pass } " looks weird. Shoudn't this function be implemented differently.
```
/// ideally this check is in separated function
if !(REDEEM_METHOD === "reload" || REDEEM_METHOD === "no-reload") {
  throw new Error();
}

... function is implemented here...

```

Reply:
Definitely opened to suggestions here, I'd like to avoid throwing an error as the `no-reload` method is valid, there is just no work for it here.

The `// pass` syntax is what allows it to pass `eslint`

Reply:
Took a stab at it, generalised an `isValidRedeemMethod`, that checks configs for valid methods, let me know what you think.